mllib/src/main/scala/org/apache/spark/ml/util/DatasetUtils.scala

```diff
@@ -17,16 +17,57 @@
 
 package org.apache.spark.ml.util
 
-import org.apache.spark.ml.linalg.{Vector, Vectors, VectorUDT}
+import org.apache.spark.ml.linalg._
 import org.apache.spark.mllib.linalg.{Vector => OldVector, Vectors => OldVectors}
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.{Column, Dataset, Row}
-import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.functions.{col, lit, udf}
 import org.apache.spark.sql.types.{ArrayType, DoubleType, FloatType}
 
 
 private[spark] object DatasetUtils {
 
+  private[ml] def getBinaryLabelCol(labelCol: String) = {
+    checkBinaryLabel(col(labelCol).cast(DoubleType))
+  }
+
+  private[ml] def getNonNegativeWeightCol(weightCol: Option[String]) = weightCol match {
+    case Some(w) if w.nonEmpty => checkNonNegativeWeight(col(w).cast(DoubleType))
+    case _ => lit(1.0)
+  }
+
+  private[ml] def getNonNanVectorCol(featuresCol: String) = {
+    checkNonNanVector(col(featuresCol))
+  }
+
+  private def checkBinaryLabel = udf {
```

Comment:
Rather than use UDFs, I wonder if it's faster to just check whether isNaN() is true for any values using Spark SQL functions? likewise checking for 0/1, etc.

Reply:
good idea, will have a try